guides/search.md

```diff
@@ -0,0 +1,294 @@
+let page_2_body = page_2.json::<Value>().await?;
+println!("{}", page_2_body["hits"]["hits"].as_array().unwrap());
+
+page_3 = client
+  .search(SearchParts::Index(&["movies"]))
+  .size(2)
+  .body(json!({
+    "query": {
+      "match": {
+        "title": "dark knight"
+      }
+    },
+    "sort": [
+      {
+        "year": {
+          "order": "asc"
+        }
+      }
+    ],
+    "search_after": page_2_body[page_2_body.len()-1]["sort"]
+  }))
+  .send()
+  .await?;
+let page_3_body = page_3.json::<Value>().await?;
+println!("{}", page_3_body["hits"]["hits"].as_array().unwrap());
+```
+
+### Pagination with scroll
+
+When retrieving large amounts of non-real-time data, you can use the `scroll` parameter to paginate through the search results.
+
+```rust
+// Create a point in time
+pit = client.create_pit(json!({
+  "index": "movies",
+  "keep_alive": "1m"
+}));
+pit_body = pit.json::<Value>().await?;
+
+// Include pit info in the search body
+pit_search_body = json!({
+  "query": {
+    "match": {
+      "title": "dark knight"
+    }
+  },
+  "sort": [
+    {
+      "year": {
+        "order": "asc"
+      }
+    }
+  ],
+  "pit": {
+    "id": pit_body["pit_id"],
+    "keep_alive": "1m"
+  }
+});
+
+// Get the first 3 pages of results
+page_1 = client
+  .search(SearchParts::Index(&["movies"]))
+  .size(2)
+  .body(pit_search_body)
+  .send()
+  .await?
+
+let page_1_body = page_1.json::<Value>().await?;
+println!("{}", page_1_body["hits"]["hits"].as_array().unwrap());
+
+page_2 = client
+  .search(SearchParts::Index(&["movies"]))
+  .size(2)
+  .body(json!({
+    "query": {
+      "match": {
+        "title": "dark knight"
+      }
+    },
+    "sort": [
+      {
+        "year": {
+          "order": "asc"
+        }
+      }
+    ],
+    "pit": {
+      "id": pit_body["pit_id"],
+      "keep_alive": "1m"
+    },
+    "search_after": page_1_body[page_1_body.len() - 1]["sort"]
+  }))
+  .send()
+  .await?
+let page_2_body = page_2.json::<Value>().await?;
+println!("{}", page_2_body["hits"]["hits"].as_array().unwrap());
+
+page_3 = client
+  .search(SearchParts::Index(&["movies"]))
+  .size(2)
+  .body(json!({
+    "query": {
+      "match": {
+        "title": "dark knight"
+      }
+    },
+    "sort": [
+      {
+        "year": {
+          "order": "asc"
+        }
+      }
+    ],
+    "pit": {
+      "id": pit_body["pit_id"],
+      "keep_alive": "1m"
+    },
+    "search_after": page_2_body[page_2_body.len() - 1]["sort"]
+  }))
+  .send()
+  .await?
+let page_3_body = page_3.json::<Value>().await?;
+println!("{}", page_3_body["hits"]["hits"].as_array().unwrap());
+
+// Print out the titles of the first 3 pages of results
+page_1_body.iter().map(|hit| println!("{}", hit["_source"]["title"]));
+page_2_body.iter().map(|hit| println!("{}", hit["_source"]["title"]));
+page_3_body.iter().map(|hit| println!("{}", hit["_source"]["title"]));
+
+// Delete the point in time
+client.delete_pit().body(json!({"pit_id": pit_body["pit_id"]}));
```

Comment:
I noticed as part of this that the PIT apis in the Rust client aren't quite right, so I'm going to make a PR fixing them that'll need to go in first, but in the meantime I've changed your code to what they should become. You should use `.for_each` when printing result titles rather than `.map` as `map` is lazy. Otherwise general indentation and tidiness issues similar to above.

```suggestion
// Create a point in time
let pit_id = client
    .create_pit(CreatePITParts::Index(&["movies"]))
    .keep_alive("1m")
    .send()
    .await?
    .json::<Value>()
    .await?["pit_id"]
    .to_string();

// Get the first 3 pages of results, including pit info in the search body
let page_1 = client
    .search(SearchParts::Index(&["movies"]))
    .size(2)
    .body(json!({
        "query": {
            "match": {
                "title": "dark knight"
            }
        },
        "sort": [
            {
                "year": {
                    "order": "asc"
                }
            }
        ],
        "pit": {
            "id": pit_id,
            "keep_alive": "1m"
        }
    }))
    .send()
    .await?
    .json::<Value>()
    .await?;
let page_1_hits = page_1["hits"]["hits"].as_array().unwrap();
println!("{:?}", page_1_hits);

let page_2 = client
    .search(SearchParts::Index(&["movies"]))
    .size(2)
    .body(json!({
        "query": {
            "match": {
                "title": "dark knight"
            }
        },
        "sort": [
            {
                "year": {
                    "order": "asc"
                }
            }
        ],
        "pit": {
            "id": pit_id,
            "keep_alive": "1m"
        },
        "search_after": page_1_hits[page_1_hits.len() - 1]["sort"]
    }))
    .send()
    .await?
    .json::<Value>()
    .await?;
let page_2_hits = page_2["hits"]["hits"].as_array().unwrap();
println!("{:?}", page_2_hits);

let page_3 = client
    .search(SearchParts::Index(&["movies"]))
    .size(2)
    .body(json!({
        "query": {
            "match": {
                "title": "dark knight"
            }
        },
        "sort": [
            {
                "year": {
                    "order": "asc"
                }
            }
        ],
        "pit": {
            "id": pit_id,
            "keep_alive": "1m"
        },
        "search_after": page_2_hits[page_2_hits.len() - 1]["sort"]
    }))
    .send()
    .await?
    .json::<Value>()
    .await?;
let page_3_hits = page_3["hits"]["hits"].as_array().unwrap();
println!("{:?}", page_3_hits);

// Print out the titles of the first 3 pages of results
page_1_hits
    .iter()
    .for_each(|hit| println!("{}", hit["_source"]["title"]));
page_2_hits
    .iter()
    .for_each(|hit| println!("{}", hit["_source"]["title"]));
page_3_hits
    .iter()
    .for_each(|hit| println!("{}", hit["_source"]["title"]));

// Delete the point in time
client
    .delete_pit(DeletePITParts::None)
    .body(json!({ "pit_id": pit_id }))
    .send()
    .await?;
```